plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -712,6 +748,60 @@ private void finishOptimize(IcebergTableExecuteHandle executeHandle, Collection<
         transaction = null;
     }
 
+    @Override
+    public void executeTableExecute(ConnectorSession session, ConnectorTableExecuteHandle tableExecuteHandle)
+    {
+        IcebergTableExecuteHandle executeHandle = (IcebergTableExecuteHandle) tableExecuteHandle;
+        switch (executeHandle.getProcedureId()) {
+            case VACUUM:
+                executeVacuum(session, executeHandle);
+                return;
+            default:
+                throw new IllegalArgumentException("Unknown procedure '" + executeHandle.getProcedureId() + "'");
+        }
+    }
+
+    private void executeVacuum(ConnectorSession session, IcebergTableExecuteHandle executeHandle)
+    {
+        IcebergVacuumHandle vacuumHandle = (IcebergVacuumHandle) executeHandle.getProcedureHandle();
+        Table table = catalog.loadTable(session, executeHandle.getSchemaTableName());
+        long retentionThreshold = vacuumHandle.getRetentionThreshold().toMillis();
+        long expireTimestampMillis = System.currentTimeMillis() - retentionThreshold;
+        table.expireSnapshots()
+                .expireOlderThan(expireTimestampMillis)
+                .cleanExpiredFiles(true)
+                .commit();
+        deleteOrphanFiles(table, session, executeHandle.getSchemaTableName(), expireTimestampMillis);
+    }
+
+    private void deleteOrphanFiles(Table table, ConnectorSession session, SchemaTableName schemaTableName, long expireTimestamp)
+    {
+        Set<String> validDataFilePaths = Streams.stream(table.snapshots())
+                .map(snapshot -> snapshot.snapshotId())
```

Comment:
```suggestion
                .map(snapshot -> snapshot.snapshotId())
```

Reply:
nit: cosmetic change `snapshot -> snapshot.snapshotId()` -> `Snapshot::snapshotId`

Reply:
now I see the difference